### DISCUSSION TITLE
SQL DB plugin and AWS Athena

### DISCUSSION THREAD

**Discussion by jessicaparks** [nextflow-io/nf-sqldb#5] [General]
Is it possible to use the SQL DB plugin to query data from Athena? If not, is this a feature you're considering adding in the future?

**Comment by pditommaso** [comment]
This plugin could support any DB having a [JDBC](https://docs.oracle.com/javase/tutorial/jdbc/basics/index.html) interface. AFAIK Athena does not provide one. 

Therefore it should require an Athena native API implementation, which would be very interesting to have tho not have a concrete plan in the short term

**Reply by jessicaparks** [reply]
Looks like this JDBC driver can be used to connect to Athena: https://docs.aws.amazon.com/athena/latest/ug/connect-with-jdbc.html. Do you think this would work?

**Reply by pditommaso** [reply]
This is definitely useful. Now that you are mentioning it, I remember I have seen this driver in the past but was a commercial product, instead it looks now it's has become an open-source project. Therefore it can be included in this this plugin distribution

**Reply by pditommaso** [reply]
@abhi18av keen to give a try to this? 

@jessicaparks do you have any example of a public data source we could use as a test case?

**Reply by abhi18av** [reply]
Sure @pditommaso!

Hi @jessicaparks , yes this definitely seems interesting! 

I'll make a push to integrate the SRA query capability via AWS Athena to the https://github.com/nextflow-io/nf-sraql plugin. 

I'll ping here as soon as its ready to be tested - would be great to receive your feedback!

**Reply by pditommaso** [reply]
@abhi18av being a JBDC for Athena, I think the goal is the add the library in this project and make some tests to verify it works properly.

**Reply by abhi18av** [reply]
Ah gotcha - mixed the two projects here due to the overlap with AWS Athena!

**Reply by abhi18av** [reply]
@pditommaso  regarding this - the jar is not published on Maven. The driver is contained in a Zip file with the JARs for JDBC-4.1 and JDBC-4.2 https://s3.amazonaws.com/athena-downloads/drivers/JDBC/SimbaAthenaJDBC-2.0.25.1002/SimbaAthenaJDBC-2.0.25.1002.zip

I have a couple questions

1. Does the license here https://s3.amazonaws.com/athena-downloads/drivers/JDBC/SimbaAthenaJDBC-2.0.25.1001/doc/LICENSE.txt look good, to be distributed along with this plugin?

2. As per https://www.baeldung.com/gradle-dependencies-local-jar, it seems it's possible to add the jar directly via `Filesystem`. Then for distribution, we add the `JDBC 4.2` JAR to `lib` folder and then compile and distribute  with the plugin?

**Reply by abhi18av** [reply]
I've added the draft work here https://github.com/nextflow-io/nextflow/pull/2500 , eager to hear your thoughts.

**Reply by jorgeaguileraseqera** [reply]
related

https://github.com/nextflow-io/nextflow/pull/2890
